### PR TITLE
FairFindPackage2: Improve implicit dependency discovery

### PR DESCRIPTION
**Pull request by dennisklein**
Enables the following:

```cmake
find_package(FairCMakeModules 0.2 REQUIRED)
include(FairFindPackage2)
find_package2(PRIVATE FairLogger VERSION 1.6 REQUIRED)
find_package2(PRIVATE FairMQ VERSION 1.4 REQUIRED)
(...)
find_package2_implicit_dependencies()
```

### CODE REVIEW

File: cmake/TestLib.cmake [hunk 1]
```diff
@@ -62,3 +62,163 @@ function(assert_var_equal varname expected)
...
+  if(ARGS_DIR)
+    set(__dir__ ${ARGS_DIR})
+  else()
+    set(__dir__ ${CMAKE_CURRENT_BINARY_DIR})
+  endif()
+
+  set(__pkgdir__ "${__dir__}/${pkgname}-${version}")
+  file(MAKE_DIRECTORY ${__pkgdir__})
+
+  foreach(dep IN LISTS ARGS_DEPS)
```
Comment (ChristianTackeGSI) [inline cmake/TestLib.cmake:201]: Did you consider to clear things here like this?
```cmake
set(__PROJECT_INTERFACE_PACKAGE_DEPENDENCIES)
```

File: cmake/TestLib.cmake [hunk 2]
```diff
@@ -62,3 +62,163 @@ function(assert_var_equal varname expected)
...
+    string(REGEX MATCH "([^:])+(:(.*))*" _MATCH "${dep}")
```
Comment (ChristianTackeGSI) [inline cmake/TestLib.cmake]: Please recheck, whether `([^:]+)` works better?

**Review by ChristianTackeGSI — DISMISSED** [review]
Two comments. Please check.

Rest looks great. I like that!

File: cmake/TestLib.cmake (hunk 1 above)
Comment (dennisklein) [reply cmake/TestLib.cmake:201]: I added it, thx!

File: cmake/TestLib.cmake (hunk 2 above)
Comment (dennisklein) [reply cmake/TestLib.cmake]: Good catch, changed.

**Review by ChristianTackeGSI — APPROVED** [review]

**Pull request merged**